st2common/tests/unit/services/test_packs.py

```diff
@@ -0,0 +1,193 @@
+
+
+class DeleteActionFilesErrorTest(unittest2.TestCase):
+    def setUp(self):
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        # creating test entry_point file in dummy pack
+        with open(entry_point, "w") as f:
+            f.write("# entry point file to be removed")
+
+        # creating test metadata file in dummy pack
+        with open(metadata_file, "w") as f:
+            f.write("# metadata file to be removed")
+
+    def tearDown(self):
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        # removing test entry_point and metadata file in dummy pack
+        os.remove(entry_point)
+        os.remove(metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_entry_point_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_permission_error_to_remove_resource_metadata_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = PermissionError("No permission to delete file from disk")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(metadata_file))
+
+        # asserting delete_action_files_from_pack function raises PermissionError
+        with self.assertRaises(PermissionError):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
+
+    @mock.patch.object(os, "remove")
+    def test_exception_to_remove_resource_entry_point_file(self, remove):
+
+        entry_point = os.path.join(TEST_PACK_PATH, "actions", "test_entry_point.py")
+        metadata_file = os.path.join(TEST_PACK_PATH, "actions", "test_metadata.yaml")
+
+        remove.side_effect = Exception("Another exception occured")
+
+        # asserting metadata files exists
+        self.assertTrue(os.path.exists(entry_point))
+
+        # asserting delete_action_files_from_pack function raises exception
+        with self.assertRaises(Exception):
+            delete_action_files_from_pack(TEST_PACK, entry_point, metadata_file)
```

Comment:
How can we tell this is the error for the entry point file?

Reply:
This looks resolved.